### PR TITLE
Merge File Task So PowerShell Modules Can Optimize Import Speed

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: Test/MergeFile.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+}
+
+Describe 'MergeFile.when separator and files are binary' {
+    It 'should concatenate files with separator' {
+        Init
+        [byte[]]$content = Get-RandomByte
+        GivenFile 'one.txt' -WithContent $content 
+        [byte[]]$content2 = Get-RandomByte
+        GivenFile 'two.txt' -WithContent ($content2)
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt' -WithSeparator ([byte[]]@( 28 ))
```
Comment (DecoyJoe) [inline Test/MergeFile.Tests.ps1]: We should have a test to ensure that the `TextSeparator` property takes precedence over the `BinarySeparator` property.

File: Test/MergeFile.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    It 'should separate file contents with separator' {
+        Init
+        GivenFile 'one.txt' 
+        GivenFile 'two.txt' 
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt' 
+        ThenFile 'merged.txt' -HasContent ('')
+    }
+}
+
+Describe 'MergeFile.when files are binary' {
```
Comment (DecoyJoe) [inline Test/MergeFile.Tests.ps1]: Maybe also a test for files text file that are not `UTF-8` encoded? E.g. `UTF-8 with BOM` or `UTF-16`

File: whiskey.yml [hunk 3]
```diff
@@ -38,11 +38,20 @@ Build:
     Path:
     - Test\about_Whiskey_Tasks.Tests.ps1
 
+# Dot-sourcing a lot of files when importing a module varys from expensive to really expensive. Merge them into the module manifest.
+- MergeFile:
+    Path:
+    - Whiskey\Functions\*.ps1
+    - Whiskey\Tasks\*.ps1
```
[the remark this answers is not in the document]
Comment (splatteredbits) [reply whiskey.yml]: Yes. Hence the reason I wrote the Glob PowerShell module. I just haven't thought of a good way to add support. Maybe through the new path validation attribute?

File: Test/Resolve-WhiskeyDotnetSdkVersion.Tests.ps1 [hunk 4]
```diff
@@ -67,7 +68,6 @@ function ThenResolvedLatestLTSVersion
     $ltsVersion = $Matches[1]
 
     $resolvedVersion | Should -HaveCount 1 -Because 'it should only return one version'
-    $resolvedVersion | Should -Be $ltsVersion -Because 'it should resolve the latest LTS version'
```
[the remark this answers is not in the document]
Comment (splatteredbits) [reply Test/Resolve-WhiskeyDotnetSdkVersion.Tests.ps1]: Bad merge.

File: Test/MergeFile.Tests.ps1 (hunk 1 above)
Comment (splatteredbits) [reply Test/MergeFile.Tests.ps1]: I added a failure condition if both are provided.

File: Test/MergeFile.Tests.ps1 (hunk 2 above)
Comment (splatteredbits) [reply Test/MergeFile.Tests.ps1]: This test exists to ensure that the task doesn't use or switch to using a TextReader or the `IO.File.*AllText` . It should operate on files without changing a single byte in them. Adding a test with a UTF-16-encoded file wouldn't do more than this test does.

File: whiskey.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (DecoyJoe) [reply whiskey.yml]: Yup, that's exactly how I was thinking of doing it, making the `ValidatePath` attribute support Glob patterns for any paths it validates.

File: Whiskey/en-US/about_Whiskey_MergeFile_Task.help.txt [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+DESCRIPTION
+    The `MergeFile` task takes one or more source files and merges them together into a destination file. Pass the list of source files to merge to the `Path` property (wildcards are supported). Pass the path to the destination file where the files should be merged to the `DestinationPath` property. The destination file will be created if it doesn't exist. If the destination file already exists, its contents will be preserved. To clear the contents of the destination file before merging in other files, set the `Clear` property to `true`.
+
+    By default, the source files are left in place. If you'd like to delete the source files as they are merged into the destination file, set the `DeleteSourceFiles` property to `true`.
+
+    Files are concateneated as-is. If you want to ensure each file is separated from other files appropriately, set either the `TextSeparator` property to a string to use to separate each file, or the `BinarySeparator` property to an array of bytes (i.e. numbers between 0 and 255). After merging each file, the task adds the separator. If both `TextSeparator` and `BinarySeparator` are given, `TextSeparator` takes precedence.
```
Comment (DecoyJoe) [inline Whiskey/en-US/about_Whiskey_MergeFile_Task.help.txt]: This is no longer valid: `If both TextSeparator and BinarySeparator are given, TextSeparator takes precedence.`

File: Whiskey/en-US/about_Whiskey_MergeFile_Task.help.txt [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+PROPERTY
+    * `Path` (*mandatory*): a list of one or more paths to merge. Wildcards are supported. Must be to files under the current build root.
+    * `DestinationPath (*mandatory*): the path to the file into which the files will be merged. Must be under the build root. Will be created if it doesn't exist. Existing contents will be preserved. Set the `Clear` property to `true` to remove any existing contents before merging any files.
+    * `DeleteSourceFiles`: if set to `true`, will delete each source file after is is merged into the destination file. The default value is `false`.
+    * `TextSeparator`: a string of text to use to separate each file's content when merging. We recommend `$(NewLine)` when merging text files.
+    * `BinarySeparator`: a list of bytes to use to separate each file's content when merging. Each item in the list must be a number between 0 and 255. 
+    * `Clear`: by default, the task leaves to destination file's content in place. To clear the destination file before merging files, set this property to `true`.
```
Comment (DecoyJoe) [inline Whiskey/en-US/about_Whiskey_MergeFile_Task.help.txt]: Typo: `leaves to destination` should be `leaves the destination`

**Pull request merged**